frontend/src/components/auth/RouteGuard.tsx

```
@@ -39,9 +39,29 @@ export const RouteGuard: React.FC<RouteGuardProps> = ({
-  if (!requireAuth && supabaseUser && serverUser) {
-    return <Navigate to="/home" replace />;
+  // For public routes, handle authenticated users
+  if (!requireAuth && supabaseUser) {
+    // If user is authenticated but needs to complete profile
+    if (isNewUser || !serverUser) {
+      console.log('🔀 RouteGuard: Redirecting to complete profile');
+      return <Navigate to="/complete-profile" replace />;
+    }
+    
+    // If user is authenticated and has a profile, redirect to their profile
+    if (serverUser && serverUser.username) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)